xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,165 @@
+from dataclasses import dataclass
+import argparse
+import difflib
+import json
+import subprocess
+import re
+import os
+from typing import List, Optional
+
+# Path to compiled C++ executable
+xml_converter: str = "../build/xml_converter"
+json_file_path: str = "../test_cases/test_expected_outputs.json"
+
+
+@dataclass
+class XMLConverter_arguments:
+    arg_input_xml: str = "--input-taco-path"
+    arg_output_xml: str = "--output-taco-path"
+    arg_input_proto: str = "--input-waypoint-path"
+    arg_output_proto: str = "--output-waypoint-path"
+    arg_split_proto: str = "--output-split-waypoint-path"
```

Comment:
This is being used as a static dictionary for a magic string. A better convention here is to just make these top level constant variables
```
arg_input_xml: Final[str] = "--input-taco-path"
...
```
no need to make a dataclass for static data. These are also only used once and not a value that a programmer would really need to care about changing too so you might not need them at all.

Reply:
They could be. But Final is not important enough for me to prevent a merge over.